2018/Alekseev/FootballLeague/Football_League/Match.cs

```diff
@@ -1,46 +1,47 @@
-﻿namespace Football_League
+﻿using System;
+
+namespace Football_League
 {
+    [Serializable]
     public class Match
     {
-        private readonly Contestant _playerOne;
-        private readonly Contestant _playerTwo;
-        public Contestant Winner;
-        public Contestant Loser;
+        public Contestant Winner, Loser;
+        public Match NextMatch, NextRoundMatch;
+        public Contestant PlayerOne, PlayerTwo;
 
+        public Match()
+        {
+            
+        }
         public Match(Contestant first = null, Contestant second = null)
         {
-            _playerOne = first;
-            _playerTwo = second;
+            PlayerOne = first;
+            PlayerTwo = second;
         }
 
-        public string PlayerOne => _playerOne.Name;
-
-        public string PlayerTwo => _playerTwo.Name;
-
-        public bool AutoWin()
+        public Contestant PickWinner()
         {
-            if (_playerTwo == null)
+            if (PlayerTwo == null && PlayerOne != null)
+            {
+                Winner = PlayerOne;
+                return Winner;
+            }
+            if (ConsoleWorker.ChooseMatchWinner(this) == 1)
             {
-                Winner = _playerOne;
-                Loser = null;
-                return true;
+                Winner = PlayerOne;
+                Loser = PlayerTwo;
             }
-            if (_playerOne == null)
+            else
             {
-                Loser = null;
-                Winner = _playerTwo;
-                return true;
+                Winner = PlayerTwo;
+                Loser = PlayerOne;
             }
-            return false;
+            return Winner;
```

Comment:
Don't think we need a return of Winner there. This value is stored in a public field anyway. Maybe it's not convinient to use down there, IDK.